drivers/serial/uart_nrfx_uart.c

```diff
@@ -407,24 +454,41 @@ static int uart_nrfx_init(struct device *dev)
 }
 
 static const struct uart_driver_api uart_nrfx_uart_driver_api = {
-	.poll_in          = uart_nrfx_poll_in,          /** Console I/O function */
-	.poll_out         = uart_nrfx_poll_out,         /** Console I/O function */
-	.err_check        = uart_nrfx_err_check,        /** Console I/O function */
+	/** Console I/O function */
+	.poll_in          = uart_nrfx_poll_in,
+	/** Console I/O function */
+	.poll_out         = uart_nrfx_poll_out,
+	/** Console I/O function */
+	.err_check        = uart_nrfx_err_check,
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
-	.fifo_fill        = uart_nrfx_fifo_fill,        /** IRQ FIFO fill function */
-	.fifo_read        = uart_nrfx_fifo_read,        /** IRQ FIFO read function */
-	.irq_tx_enable    = uart_nrfx_irq_tx_enable,    /** IRQ transfer enabling function */
-	.irq_tx_disable   = uart_nrfx_irq_tx_disable,   /** IRQ transfer disabling function */
-	.irq_tx_ready     = uart_nrfx_irq_tx_ready,     /** IRQ transfer ready function */
-	.irq_rx_enable    = uart_nrfx_irq_rx_enable,    /** IRQ receiver enabling function */
-	.irq_rx_disable   = uart_nrfx_irq_rx_disable,   /** IRQ receiver disabling function */
-	.irq_tx_complete  = uart_nrfx_irq_tx_complete,  /** IRQ transfer complete function */
-	.irq_rx_ready     = uart_nrfx_irq_rx_ready,     /** IRQ receiver ready function */
-	.irq_err_enable   = uart_nrfx_irq_err_enable,   /** IRQ error enabling function */
-	.irq_err_disable  = uart_nrfx_irq_err_disable,  /** IRQ error disabling function */
-	.irq_is_pending   = uart_nrfx_irq_is_pending,   /** IRQ pending status function */
-	.irq_update       = uart_nrfx_irq_update,       /** IRQ interrupt update function */
-	.irq_callback_set = uart_nrfx_irq_callback_set, /** Set the callback function */
+	/** IRQ FIFO fill function */
+	.fifo_fill        = uart_nrfx_fifo_fill,
+	/** IRQ FIFO read function */
+	.fifo_read        = uart_nrfx_fifo_read,
+	/** IRQ transfer enabling function */
+	.irq_tx_enable    = uart_nrfx_irq_tx_enable,
+	/** IRQ transfer disabling function */
+	.irq_tx_disable   = uart_nrfx_irq_tx_disable,
+	/** IRQ transfer ready function */
+	.irq_tx_ready     = uart_nrfx_irq_tx_ready,
+	/** IRQ receiver enabling function */
+	.irq_rx_enable    = uart_nrfx_irq_rx_enable,
+	/** IRQ receiver disabling function */
+	.irq_rx_disable   = uart_nrfx_irq_rx_disable,
+	/** IRQ transfer complete function */
+	.irq_tx_complete  = uart_nrfx_irq_tx_ready,
```

Comment:
This looks confusing (seems to be a mistake).
It should be at least commented (that a common function is used for two API entries, since our hardware don't provide these two statuses separately). And I would suggest to include `_complete` in the function name as well.